src/assisted_test_infra/test_infra/controllers/node_controllers/terraform_controller.py

```diff
@@ -61,8 +61,8 @@ def _get_disk_encryption_appliance(self):
             return {}
 
         assert (
-            self._entity_config.disk_encryption_mode == consts.DiskEncryptionMode.TPM_VERSION_2
-        ), "Currently only supporting TPMv2"
+            self._entity_config.disk_encryption_mode in consts.DiskEncryptionMode.all()
+        ), f"{self._entity_config.disk_encryption_mode} is not a supported disk encryption mode"
 
         master_vtpm2 = worker_vtpm2 = False
 
```

Comment:
One more change that you need to do:
```
        master_vtpm2 = worker_vtpm2 = False

        if self._entity_config.disk_encryption_mode == consts.DiskEncryptionMode.TPM_VERSION_2:
            if self._entity_config.disk_encryption_roles == consts.DiskEncryptionRoles.ALL:
                master_vtpm2 = worker_vtpm2 = True
            elif self._entity_config.disk_encryption_roles == consts.DiskEncryptionRoles.MASTERS:
                master_vtpm2 = True
            elif self._entity_config.disk_encryption_roles == consts.DiskEncryptionRoles.WORKERS:
                worker_vtpm2 = True
```

otherwise it's enabling tpm on tang mode